lib/Jsrt/JsrtHelper.cpp

```diff
@@ -49,3 +49,121 @@ void JsrtCallbackState::ObjectBeforeCallectCallbackWrapper(JsObjectBeforeCollect
     JsrtCallbackState scope(reinterpret_cast<ThreadContext*>(threadContext));
     callback(object, callbackState);
 }
+
+#if defined(CHAKRA_STATIC_LIBRARY) || !defined(_WIN32)
+    #include "Core/ConfigParser.h"
+    #include "Base/ThreadBoundThreadContextManager.h"
+
+#ifndef _WIN32
+    #include <pthread.h>
+    static pthread_key_t s_threadLocalDummy;
+#endif
+
+#ifndef __APPLE__
+    #if defined(_MSC_VER) && _MSC_VER <= 1800 // VS2013?
+        // Do not use this as solution wide!
+        // This is only for internal static library only
+        #define THREAD_LOCAL __declspec(thread)
+    #else // VS2015+, linux Clang etc.
+        #define THREAD_LOCAL thread_local
+    #endif // VS2013?
+#else // __APPLE__
+    #define THREAD_LOCAL _Thread_local
+#endif
+
+    static THREAD_LOCAL bool s_threadWasEntered = false;
+
+    _NOINLINE void DISPOSE_CHAKRA_CORE_THREAD(void *_)
+    {
+        free(_);
+        ThreadBoundThreadContextManager::DestroyContextAndEntryForCurrentThread();
+    }
+
+    _NOINLINE bool InitializeProcess()
+    {
+#if !defined(_WIN32)
+        pthread_key_create(&s_threadLocalDummy, DISPOSE_CHAKRA_CORE_THREAD);
+#endif
+
+#if defined(CHAKRA_STATIC_LIBRARY)
+#ifndef _WIN32
```

Comment:
A lot of this looks copy-pasted from AttachProcess in ChakraCoreDllFunc.cpp, yes? Can you add a comment in both places indicating that updating one should update an other, and a TODO comment to consolidate?